models/_generate_layers.py

```diff
@@ -496,6 +497,50 @@ def collect_layers():
   return ls
 
 
+def get_super_call_params(layer: LayerBase) -> str:
+  """
+  Inspects the given layer in RETURNN, extracts the super call from there, removes unwanted parameters
+  and then returns a string that is ready to be used in the super call for the _generated_layers file.
+  :param layer: Layer to be inspected
+  :return: string with super call parameters
+  """
+  # get code as string list
+  code = inspect.getsource(layer.__init__).splitlines()
+
+  # get list of lines starting with super and take the first one
+  lines = [line.strip() for line in code if line.strip().startswith("super")]
+  if not lines:
+    return ''
+
+  # reformat the super call to extract what we need
+  # get the first line which contains super to get the super call
+  super_call = lines[0]  # super_call="super(ChoiceLayer, self).__init__(beam_size=beam_size, search=search, **kwargs)"
+  # only keep the part after init of super_call
+  call = super_call.split(".")[1]  # call = "__init__(beam_size=beam_size, search=search, **kwargs)"
+  assert call.startswith("__init__")
+  assert call.endswith(")")
+  # remove "__init__(" and ") from the call string
+  call_pruned = call[len("__init__("):-len(")")]  # call pruned = "beam_size=beam_size, search=search, **kwargs"
+
+  # get list of tuples for parameter with (param_name, value)
+  tup_ls = [x.split("=") for x in call_pruned.split(",")]
+
+  # ignore **kwargs when excluding
```

Comment:
Again, do not hard code some assumptions without checking them.

But also, this could have been easier directly in the code above:
```
tup_ls = [x.split("=") for x in call_pruned.split(",") if x != "**kwargs"]
```